### PR TITLE
Initial import of MockTracer.

### PR DESCRIPTION
**Pull request by carlosalberto**

### CODE REVIEW

File: opentracing/mocktracer/context.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+from __future__ import absolute_import
+
+import opentracing
+
+
+class SpanContext(opentracing.SpanContext):
+    """SpanContext satisfies the opentracing.SpanContext contract.
+
+    trace_id and span_id are uint64's, so their range is [0, 2^64).
```
Comment (yurishkuro) [inline opentracing/mocktracer/context.py]: `(0, 2^64)` - probably don't want 0 as a valid value

File: opentracing/mocktracer/util.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+def generate_id():
+    global _fork_guard_pid
+
+    # Microbenchmarks suggest that os.getpid() takes less than 0.1 microsecond.
+    pid = os.getpid()
+    if (_fork_guard_pid == 0) or (_fork_guard_pid != pid):
+        _fork_guard_pid = pid
+        guid_rng.seed(int(1000000 * time.time()) ^ pid)
+    return guid_rng.getrandbits(64) - 1
```
Comment (yurishkuro) [inline opentracing/mocktracer/util.py]: would be good to ensure 0 is never returned

File: opentracing/mocktracer/context.py (hunk 1 above)
Comment (carlosalberto) [reply opentracing/mocktracer/context.py]: :)

File: opentracing/mocktracer/util.py (hunk 2 above)
Comment (carlosalberto) [reply opentracing/mocktracer/util.py]: Sure :)

**Comment by carlosalberto** [comment]
Hey @yurishkuro - so since we don't need a complex id generation phase, I went and did what the `MockTracer` for Java does: a simple locked, auto-increasing counter to get the next id.

I think this should work just fine for our mocking layer (making also sure `0` is not used as a possible value for `trace_id` nor `span_id`).

Let me know ;)

**Comment by yurishkuro** [comment]
Sgtm

**Pull request merged**